src/SparseEnsembleKalmanInversion.jl

```diff
@@ -158,6 +154,10 @@ function update_ensemble!(
         )
     end
 
+    # store new parameters (and model outputs)
+    push!(ekp.u, DataContainer(u, data_are_columns = true))
+    push!(ekp.g, DataContainer(g, data_are_columns = true))
```

Comment:
Now that this part is moved after the sparsity modification, let's encapsulate L130-L155 along with the rest of update equations to a separate function called `sparse_eki_update`. You can see an example as `eki_update` in `EnsembleKalmanInversion.jl`. Then, let's call this function through the failsafe_update. I couldn't do it in the FailureHandler PR because the equations were split by these `push!` commands.

Reply:
Done. Feel free to let me know if any other modification is needed.